src/CloudNative.CloudEvents/CloudEventsSpecVersion.cs

```diff
@@ -1,15 +1,146 @@
-﻿// Copyright (c) Cloud Native Foundation. 
+﻿// Copyright 2021 Cloud Native Foundation. 
 // Licensed under the Apache 2.0 license.
 // See LICENSE file in the project root for full license information.
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+using static CloudNative.CloudEvents.CloudEventAttribute;
+
 namespace CloudNative.CloudEvents
 {
-    public enum CloudEventsSpecVersion
+    // TODO: Rename to CloudEventSpecVersion? Using the plural form feels a little odd.
+
+    /// <summary>
+    /// Represents a version of the CloudEvents specification, including
+    /// the context attribute values known to that version.
+    /// </summary>
+    public sealed class CloudEventsSpecVersion
     {
-        V0_1,
-        V0_2,
-        V0_3,
-        V1_0,
-        Default = V1_0
+        /// <summary>
+        /// The attribute used to indicate the version of the CloudEvents specification being used.
+        /// </summary>
+        public static CloudEventAttribute SpecVersionAttribute { get; } = CreateRequired("specversion", CloudEventAttributeType.String, NonEmptyString);
+
+        // Populated by the constructor.
+        private static readonly List<CloudEventsSpecVersion> allVersions = new List<CloudEventsSpecVersion>();
+
+        public static CloudEventsSpecVersion Default => V1_0;
+
+        public static CloudEventsSpecVersion V1_0 { get; } = new CloudEventsSpecVersion(
+            "1.0",
+            CreateRequired("id", CloudEventAttributeType.String, NonEmptyString),
+            CreateRequired("source", CloudEventAttributeType.UriReference, NonEmptyUri),
+            CreateRequired("type", CloudEventAttributeType.String, NonEmptyString),
+            CreateOptional("datacontenttype", CloudEventAttributeType.String, Rfc2046String), // TODO: Do we want some way of adding validation that this is per RFC 2046?
+            CreateOptional("dataschema", CloudEventAttributeType.Uri, NonEmptyUri),
+            CreateOptional("subject", CloudEventAttributeType.String, NonEmptyString),
+            CreateOptional("time", CloudEventAttributeType.Timestamp, null));
+
+        /// <summary>
+        /// The ID of the spec version, in its canonical serialized form.
+        /// </summary>
+        public string VersionId { get; }
```

Comment:
Would it be worth giving an example in the docs?

Reply:
Done.